sdb/commands/nvpair/nvpair.py

```diff
@@ -0,0 +1,209 @@
+                print(f"{indent}{hex(int(val))}")
+            elif "string" in type_:
+                offset += len(val)
+                val = sdb.create_object("char *", self.data_ptr +
+                                        offset).string_().decode("utf-8")
+                print(f"{indent}{val}")
+                offset += 1  # null terminator
+
+    # obtain and print the nvpair value
+    def get_value(self) -> None:
+        type_ = (self.get_type(1) + "_t").lower()
+        if "array" in type_:
+            self.get_array_values(type_)
+            return
+        if "nvlist" in type_:
+            print("values=")
+            self.nvl.set_indent(self.nvl.indent + 4)
+            self.nvl.print_one(sdb.create_object("nvlist_t *", self.data_ptr))
+            self.nvl.set_indent(self.nvl.indent - 4)
+            return
+        print("value=", end='')
+        if "boolean_value" in type_:
+            type_ = "uint32_t"
+        elif "boolean" in type_:  # DATA_TYPE_BOOLEAN has name and no value
+            print("")
+        elif "hrtime" in type_:
+            type_ = "int64_t"
+        if "int" in type_:
+            print(self.get_value_int(type_, self.data_ptr))
+        if "string" in type_:
+            print(
+                sdb.create_object("char *",
+                                  self.data_ptr).string_().decode("utf-8"))
+        if "byte" in type_:
+            print(hex(int(self.get_value_int("unsigned char", self.data_ptr))))
+        return
+
+
+class Nvlist(sdb.SingleInputCommand):
+    """
+    Print nvlist_t
+
+    DESCRIPTION
+        Print all the nvpair_t in the passed nvlist_t.  Handle basic types,
+        array types, and nvlist_t types.  Type double is omitted as it is
+        not used in zfs.
+
+    EXAMPLE
+    Print nvlist_t of snapshot holds from the nvlist_t * pointer address:
+
+        sdb> echo 0xffff970d0ff681e0 | nvlist
+        name=monday-1 type=DATA_TYPE_UINT64 value=1633989858
+        name=monday-2 type=DATA_TYPE_UINT64 value=1633989863
+    """
+
+    names = ["nvlist"]
+    input_type = "nvlist_t *"
+    output_type = "nvlist_t *"
+    indent = 0
+
+    def set_indent(self, indent: int) -> None:
+        self.indent = indent
+
+    # nvlist iteration methods
+    @staticmethod
+    def nvlist_contains_nvp(nvl: drgn.Object, nvp: drgn.Object) -> int:
+        priv = drgn.cast("nvpriv_t *", nvl.nvl_priv)
+
+        if nvp.address_ == 0:
+            return 0
+
+        curr = priv.nvp_list
+        while not sdb.is_null(curr):
+            if curr.nvi_nvp.address_ == nvp.address_:
+                return 1
+            # pylint: disable=protected-access
+            curr = curr._nvi_un._nvi._nvi_next
+
+        return 0
+
+    @staticmethod
+    def nvlist_first_nvpair(nvl: drgn.Object) -> drgn.Object:
+        if sdb.is_null(nvl) or sdb.is_null(nvl.nvl_priv):
+            return None
+        priv = drgn.cast("nvpriv_t *", nvl.nvl_priv)
+        return priv.nvp_list.nvi_nvp
```

Comment:
It turns out that priv.nvp_list can be NULL, which is why the test output has the FaultError warning.  Will fix.